src/spec.md

```diff
@@ -0,0 +1,237 @@
+| constructor  | Create an instance of EthStorage |
+| write        | Asynchronously write data        |
+| read         | Asynchronously read data         |
+
+### FlatDirectory Class
+
+| Method Name | Description                                  |
+|-------------|----------------------------------------------|
+| constructor | Create an instance of FlatDirectory          |
+| upload      | Asynchronously upload data of arbitrary size |
+| download    | Asynchronously download data                 |
+| deploy      | Deploy a FlatDirectory contract              |
+| setDefault  | Set the default file for FlatDirectory.      |
+
+
+<p id="EthStorage"></p>
+
+## 4. EthStorage Class
+
+### Constructor
+
+**Description**: Create an instance of `EthStorage`.
+
+**Parameters**
+- `config` (object): Configuration object containing necessary settings.
+   - `rpc` (string): RPC for any evm network.
+   - `privateKey` (string): Wallet private key.
+   - `address` (string): EthStorage contract address.
+
+**Example**
+```javascript
+const config = {
+   rpc: "your_rpc",
+   privateKey: "your_private_key",
+   address: "eth_storage_address"
+};
+
+const ethStorage = new EthStorage(config);
+```
+
+### Methods
+
+#### write
+
+**Description**: Asynchronously writes data to the EthStorage network.
+
+**Parameters**
+- `key` (string): The key for the data to be written.
+- `data` (Buffer): The data to be written, its size cannot exceed the maximum value of the content that can be transferred by a blob.
+
+**Example**
+```javascript
+const dataToWrite = Buffer.from("some data");
+await ethStorage.write("dataKey", dataToWrite);
+```
+
+#### read
+
+**Description**: Read data asynchronously from the EthStorage network through key.
+
+**Parameters**
+- `key` (string): The key for the data to be read.
+- `ethStorageRpc` (string): RPC of EthStorage network, because the data is obtained from the EthStorage network.
+
+**Returns**
+- `data` (Buffer): The content.
+
+**Example**
+```javascript
+const ethStorageRpc = "https://xxx.rpc";
+const data = await ethStorage.read("example.txt", ethStorageRpc);
+```
+
+
+<p id="FlatDirectory"></p>
+
+## 5. FlatDirectory Class
+
+### Constructor
+
+**Description**: Create an instance of `FlatDirectory`.
+
+**Parameters**
+- `config` (object): Configuration object containing necessary settings.
+    - `rpc` (string): RPC for any evm network.
+    - `privateKey` (string): Wallet private key.
+    - `address` (string, optional): FlatDirectory contract address. If it does not exist, the `deploy` method can be called to create one.
+
+**Example**
+```javascript
+const config = {
+   rpc: "your_rpc",
+   privateKey: "your_private_key",
+   address: "flat_directory_address"
+};
+
+const flatDirectory = new FlatDirectory(config);
+```
+
+### Methods
+
+#### upload
+
+**Description**: Upload data of arbitrary size.
+
+**Parameters**
+- `key` (string): The key of the data.
+- `data` (Buffer): The data to be uploaded.
```

Comment:
Do we want to support the stream?

Reply:
The browser does not support stream

Reply:
https://stackoverflow.com/a/65087341/2774879